### ISSUE TITLE
Improve n8n's accessibility

### ISSUE THREAD

**Issue by webczat** [n8n-io/n8n#13872]
### Bug Description

n8n's user interface is not accessible/barely accessible. I am talking from the perspective of a screenreader/blind user.
Please improve it's accessibility.
There are lots of issues, including things as basic as being unable to actually see node connections, being unable to navigate by keyboard (tab key invoking add node is directly against normal navigation flow for example) and having no alternative to drag and drop in some/many? cases.
I'm actually surprised despite everything I can actually use it to some degree, which however doesn't make these issues minor.

### To Reproduce

n/a

### Expected behavior

n/a

### Operating System

ubuntu

### n8n Version

1.82.1

### Node.js Version

not known to me

### Database

SQLite (default)

### Execution mode

main (default)

**Comment by Joffcom** [comment]
Hey @webczat,

Typically I would use a generic reply about enhancements not being something we track here however... We do know this is a problem and something we can improve and we do plan on spending time on it soon however with so many moving parts in n8n it is not something that will happen quickly.

I will link this to the internal request to improve this so we can let you know once we have done anything around this but for now I am going to close this as isn't a bug. Let me know if you have any questions on this.

**Comment by webczat** [comment]
I decided to report it as a bug because the software is intended to be usable for it's purpose. If the user is unable to use it and application is at fault, imo it's a bug.
As for complexity, I'm aware of the fact this is a huge topic, although it can likely be split into tons of smaller things which can be done, and asome of these changes are easy to make. I just wanted to draw the attention to general problem first.

**Comment by webczat** [comment]
maybe let's start from easiest to make as opposed to biggest impact. surprisingly, I can actually, to some degree, use the software... I mean if you are really patient you can actually make workflows, it's just the worst experience ever and relying on screenreader side special features + some workarounds, and clicking random things in the hope at some time you get what you're doing. :) but it could be far worse
For that reason I wouldn't exactly call this a complete lack of accessibility. there are things that can't be used at all, so it's still something.

Probably the easiest to do is labelling buttons/actions/etc, ensuring these labels are being read, doing something with the tab key opening add node... things like this.
And for example being able to see which node is a subnode/connection of which node, the order in which my screenreader sees them looks like the order in which I add them no matter the connections, if I need to read json to make sure I get it, it's not good experience.
Also, things like expanding submenus don't usually make screenreader see I'm in the submenu, I need to do additional things to be able to use it.
Things like alternative to drag and drop, like in the "set field" node, I should be able to map fields from inputs without dragndrop.

For the complete experience, better kbd navigation/etc is probably a must, but even without it things can be improved a lot.

**Comment by sonofroot** [comment]
I believe with N8N being a German company, it will be required to comply with the EU Accessibility Act (at least with respect to N8N Cloud). Although, I'm a sighted user, we need to stop looking at accessibility issues as enhancements, and instead bugs that impact a significant user population. It's less about the fact that it's always been broken, and more the fact that accessibility (like security) is simply a functional requirement of any B2B software, not a feature.

When we de-prioritize accessibility issues, we are saying, quite literally: "Your ability to use my product isn't important to me". I don't believe this is your intention, but I struggled to empathize with this until I became friends with some incredible blind devs. 

Thanks for your hard work on the product! Looking forward to seeing the bugs get worked out as well as the new features!
